arch/arc/Kconfig

```diff
@@ -153,7 +153,7 @@ config ARC_STACK_PROTECTION
 	bool
 	default y if HW_STACK_PROTECTION
 	select ARC_STACK_CHECKING if ARC_HAS_STACK_CHECKING
-	select MPU_STACK_GUARD if (!ARC_STACK_CHECKING && ARC_MPU)
+	select MPU_STACK_GUARD if (!ARC_STACK_CHECKING && ARC_MPU && ARC_MPU_VER !=2)
```

Comment:
Is MPU stack guard still supported for non Userspace builds?
Not really clear/consistent in the code:  Kconfig implies not, include/arc/arc/arch.h implies it is?

Reply:
In the general case: MPU/MMU stack guards and user mode are orthogonal features. 

You can have one or the other, both, or neither, there should be no dependencies.